crypto/fipsmodule/rsa/rsa.c

```diff
@@ -1049,3 +1049,61 @@ int RSA_test_flags(const RSA *rsa, int flags) { return rsa->flags & flags; }
 int RSA_blinding_on(RSA *rsa, BN_CTX *ctx) {
   return 1;
 }
+
+// Performs checks equivalent to those in OpenSSL 1.x versions and 3.x non-FIPS
+// versions of |RSA_check_key|, but also add the ability to process keys with
+// public component only like AWS-LC's |RSA_check_key|.
+// Implementation checklist:
+//  + public components checks
+//  + required parameters present
+//  - p and q are prime
+//  - p * q = n
+//  - (d * e) mod (lcm(p-1, q-1)) = 1
+//  - d mod (p-1) = dmp1
+//  - d mod (q-1) = dmq1
+//  - q^-1 mod (p) = iqmp
+int wip_do_not_use_rsa_check_key(const RSA *key) {
+
+  // Every key has to have the modulus n and the public exponent e.
+  if (key->n == NULL || key->e == NULL) {
+    OPENSSL_PUT_ERROR(RSA, RSA_R_VALUE_MISSING);
+    return 0;
+  }
+
+  // Light checks on |e|, as in OpenSSL.
+  if (BN_is_one(key->e) || !BN_is_odd(key->e)) {
+    OPENSSL_PUT_ERROR(RSA, RSA_R_BAD_E_VALUE);
+    return 0;
+  }
+
+  // If the key has a private key component too, then it has to have at least
+  // the private exponent d. If d is not present, then |key| is a public key
+  // only and no more checks are performed.
```

Comment:
Why `rsa_check_public_key` is not called? Aren't we preserving those public key checks?